SimCalorimetry/HGCalSimProducers/plugins/alpaka/HGCalDigiSoaFiller.cc

```diff
@@ -0,0 +1,185 @@
+
+    /**
+       @short parameters to be used with this plugin
+    */
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    /**
+      @short steers the conversion to SOA
+    */
+    void produce(device::Event&, device::EventSetup const&) override;
+
+  private:
+    edm::EDGetTokenT<edm::SortedCollection<HGCalDataFrame> > digisCEET_, digisCEHSiT_, digisCEHSiPMT_;
+    device::EDPutToken<hgcaldigi::HGCalDigiDevice> digiProdT_;
+
+    edm::ESWatcher<HGCalElectronicsMappingRcd> cfgWatcher_;
+    //edm::ESGetToken<CellIndexer, HGCalElectronicsMappingRcd> cellIndexTkn_;
+    device::ESGetToken<CellInfo, HGCalElectronicsMappingRcd> cellTkn_;
+    edm::ESGetToken<ModuleIndexer, HGCalElectronicsMappingRcd> moduleIndexTkn_;
+    device::ESGetToken<ModuleInfo, HGCalElectronicsMappingRcd> moduleTkn_;
+  };
+
+  //
+  HGCalDigiSoaFiller::HGCalDigiSoaFiller(const edm::ParameterSet& config)
+      : digisCEET_(
+            consumes<edm::SortedCollection<HGCalDataFrame> >(config.getParameter<edm::InputTag>("eeChannelDigis"))),
+        digisCEHSiT_(
+            consumes<edm::SortedCollection<HGCalDataFrame> >(config.getParameter<edm::InputTag>("cehsiChannelDigis"))),
+        digisCEHSiPMT_(consumes<edm::SortedCollection<HGCalDataFrame> >(
+            config.getParameter<edm::InputTag>("cehsipmChannelDigis"))),
+        //cellIndexTkn_(esConsumes()),
+        cellTkn_(esConsumes()),
+        moduleIndexTkn_(esConsumes()),
+        moduleTkn_(esConsumes()) {
+    digiProdT_ = produces(config.getParameter<std::string>("digisProdName"));
+  }
+
+  //
+  void HGCalDigiSoaFiller::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("eeChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "EE"));
+    desc.add<edm::InputTag>("cehsiChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "HEfront"));
+    desc.add<edm::InputTag>("cehsipmChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "HEback"));
+    desc.add<std::string>("digisProdName", "HGC");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  //
+  void HGCalDigiSoaFiller::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
+    cfgWatcher_.check(iSetup);
+    //const CellIndexer &cellidx = iSetup.getData(cellIndexTkn_);
+    const CellInfo& cells = iSetup.getData(cellTkn_);
+    const ModuleIndexer& modidx = iSetup.getData(moduleIndexTkn_);
+    const ModuleInfo& modules = iSetup.getData(moduleTkn_);
+
+    //FIXME
+    int itSample = 2;
+
+    //read "classic" channel digis
+    auto const& digisCEE = iEvent.get(digisCEET_);
+    //auto const &digisCEHSi = iEvent.get(digisCEHSiT_);
+    //auto const &digisCEHSiPM = iEvent.get(digisCEHSiPMT_);
+
+    //fill the SoA collection in the host
+    uint32_t finaldigi_size = modidx.maxDataIdx_;
+    std::cout << "Allocating SOA with " << finaldigi_size << " entries" << std::endl;
```

Comment:
These couts should either be silenced by default or moved to MessageLogger.

Reply:
Done in https://github.com/CMS-HGCAL/cmssw/pull/137.